### PR TITLE
Yaml for Config

### PR DESCRIPTION
**Pull request by DarthHater**
Because who doesn't like yaml?!?!

Seriously though we are starting to save OSS Index config in a few tools, so why not use yaml, and a common location, so they can all pick it up. 

This pull request makes the following changes:
* Adds PyYAML
* Implements yaml load and dump in base config class
* Also moves config by default to it's own `.ossindex` folder

cc @bhamail / @DarthHater / @allenhsieh

### CODE REVIEW

**Comment by bhamail** [comment]
Digging a little more, I'm not seeing how a user should tell Jake to use a saved IQ config (`~/.iqserver/.iq-server-config` instead of the default `~/.ossindex/.oss-index-config`). The `main()` method let's you save an IQ config and then exit. But I don't see how to make Jake use that saved IQ config.

**Comment by DarthHater** [comment]
@bhamail I did not wire that up yet, but I'm planning on doing that based on if they provide an application arg or not.

**Comment by DarthHater** [comment]
@bhamail , all clear, I think!

File: jake/__main__.py [hunk 1]
```diff
@@ -32,7 +32,10 @@ def main():
...
   args = __add_parser_args_and_return()
   log = __setup_logger(args.verbose)
-  config = Config()
+  if args.application:
+    config = IQConfig()
+  else:
+    config = Config()
 
   if args.snake:
     __get_config_from_std_in(config)
```
Comment (bhamail) [inline jake/__main__.py]: Minor: down on line 43, I think the: `config = IQConfig()` of:

```
elif args.python:
    config = IQConfig()
```
is now obsolete, and should be deleted.

**Comment by DarthHater** [comment]
Yeah that config stuff is the weirdest part of the code base thus far. I'm sure we can clean it up, just haven't gotten fully there yet.

File: jake/test/test_iq_config.py [hunk 2]
```diff
@@ -36,10 +36,7 @@ def test_get_config_from_std_in(self):
     result = self.func.get_config_from_std_in()
     self.assertEqual(result, True)
     # reread stored config
-    results = self.func.get_config_from_file({"Username",
-                                              "Password",
-                                              "IQ-Server-Location"},
-                                             ".iq-server-config")
+    results = self.func.get_config_from_file(".iq-server-config")
     self.assertEqual(results["Username"], "test@me.com")
     self.assertEqual(results["Password"], "password")
```
[the remark this answers is not in the document]
Comment (DarthHater) [reply jake/test/test_iq_config.py]: Fixed in: 66bffcc

**Comment by DarthHater** [comment]
tenor

**Pull request merged**